.cascade-code/Chapman.edu/_components/left_nav/left_nav_children.vtl

```diff
@@ -0,0 +1,137 @@
+## Formerly LeftNavNoSiblings. Used for index pages.
+## New left nav format that works without a block
+
+#set ( $pagePath = $currentPage.path )
+#set ( $parentDir = $currentPage.parentFolder )
+
+## Builds the left navigation
+#macro ( buildLeftNav )
+    #buildLeftNavTitle
+    <ul>
+    #set ( $children = $parentDir.children )
+    #foreach ( $child in $children )
+        #set ( $isVisible = $child.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+
+        ## Child is a page
+        #if ( $child.assetType == 'page' && $isVisible && ($child.name != 'index') )
+            #buildChildPage($child)
+
+        ## Child is a folder
+        #elseif ( $child.assetType == 'folder' && $isVisible )
+            ## For folders, also need to check visibility of index page
+            #set ( $folderIndexPage = $_.locatePage("${child.path}/index") )
+            #set ( $visibleIndex = $folderIndexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+            ## Titles for folders are set to the name of its index page
+            #set ( $childTitle = ${_EscapeTool.xml($folderIndexPage.label)} )
+            
+            ## Only if the child folder's index page is also visible
+            #if ( $visibleIndex )
+                #set ( $visibleChildren = [] )
+                #getVisibleChildren($child.children)
+                
+                ## Needs to have a dropdown if it has visible children
+                #if ( $visibleChildren.size() > 0 )
+                    <li>
+                        <a href="${folderIndexPage.link}" class="s-f-idx">
+                            $childTitle
+                            <span class="plus">+</span>
+                        </a>
+                        #buildGrandchildren($visibleChildren)
```

Comment:
Do we have to worry about namespace collisions when using generic names like this? I'm not sure how scoping works exactly. 

I'd prefer we used more descriptive names as a rule.

Reply:
Do you mean `visibleChildren`? Or `buildGrandchildren`?

Reply:
Both potentially. With var names, I don't mind so much because I think it's reasonable to assume those are sanely scoped.

With macros, I worry a little more about namespace collisions. I'm not sure if this is a real concern. But so long as they don't get too long and crazy, I prefer more descriptive.